WP Plugin - PHPCS - Manual fixes

claytoncollie opened the pull request:
### Description of the Change
Requires #688 

These are the manual fixes for the WP plugin.

Part 2 for addressing #625 

### Changelog Entry
> Changed - Address PHP coding standards manual fixes


### Credits
Props @claytoncollie

claytoncollie committed:
Rename parameter array to be rules

claytoncollie committed:
Use wp _safe_redirect instead of wp_redirect

claytoncollie committed:
Ignore last parameter is not being used

claytoncollie committed:
remove unused parameter

claytoncollie committed:
Ignore tax query being slow

claytoncollie committed:
Ignore nonce verification warning

claytoncollie committed:
Use strict comparison for in array

claytoncollie committed:
Use wp _safe_redirect instead of wp_redirect

claytoncollie committed:
Translate string and escape output

claytoncollie committed:
Remove unused parameter and docblock

claytoncollie committed:
Remove unused parameter and docblock

claytoncollie committed:
Remove unused parameter and docblock

claytoncollie committed:
Change the admin notice to break item down into smaller pieces

claytoncollie committed:
change parameter to not be a reserved keyword

claytoncollie committed:
Ignore certain files and enforce short array syntax

claytoncollie committed:
Enforce short array syntax

nicholasio commented on wp/headless-wp/includes/classes/Links.php:
```diff
@@ -144,7 +144,7 @@ public function maybe_redirect_frontend() {
 
 			// Redirect the frontend WordPress request to the React website URL.
-			\wp_redirect( trailingslashit( esc_url_raw( $site_url ) ) . $url_request, 301 );
+			\wp_safe_redirect( trailingslashit( esc_url_raw( $site_url ) ) . $url_request, 301 );
```
In order to move to `wp_safe_redirect` we would need to add `$site_url` to `allowed_redirect_hosts`.

Is there any meaninful benefit in switching to wp_safe_redirect here?

nicholasio replied:
I'm switching back to wp_redirect for now to get this unblocked but feel free to submit a followup PR if you believe there's a real benefit in using safe_redirect here.

nicholasio commented on wp/headless-wp/includes/classes/Preview/preview.php:
```diff
@@ -38,5 +38,5 @@
 );
 
-wp_redirect( $preview_url );
+wp_safe_redirect( $preview_url );
```
Same thing as my preview comments, we would need to add the headless site url to `allowed_redirect_hosts`.

nicholasio committed:
fix: switch back to wp_redirect for now

nicholasio committed:
WP Plugin - PHPCS - Force short array syntax (#696)

### Description of the Change
Automatic linting PR to enforce short array syntax.

Require #689 

Part 3 for addressing https://github.com/10up/headstartwp/issues/625

### Changelog Entry
> Changed - Enforce short array syntax

### Credits
Props @claytoncollie
